tests/api/test_api_authors_edit.py

```diff
@@ -528,6 +527,112 @@ def test_addon_delete_owner_author(base_url, session_auth, variables):
     ), f'Actual message was {delete_owner.text}'
 
 
+@pytest.mark.serial
+@pytest.mark.create_session('staff_user')
+def test_addon_developer_role_cannot_edit_authors(base_url, session_auth, variables):
+    """An author with a developer role should not be allowed to edit existing authors,
+    like elevating their role to owners for example"""
+    addon = payloads.edit_addon_details['slug']
+    author = variables['api_post_valid_author']
+    # send the patch author requests with the developer role
+    edit_author = requests.patch(
+        url=f'{base_url}{_addon_create}{addon}/authors/{author}/',
+        headers={
+            'Authorization': f'Session {session_auth}',
+            'Content-Type': 'application/json',
+        },
+        data=json.dumps({'role': 'owner'}),
+    )
+    assert (
+        edit_author.status_code == 403
+    ), f'Actual response: {edit_author.status_code}, {edit_author.text}'
+    assert (
+        'You do not have permission to perform this action.' in edit_author.text
+    ), f'Actual message was {edit_author.text}'
+
+
+@pytest.mark.serial
+@pytest.mark.create_session('staff_user')
+def test_addon_developer_role_cannot_delete_authors(base_url, session_auth, variables):
+    """An author with a developer role should not be allowed to delete existing authors"""
+    addon = payloads.edit_addon_details['slug']
+    author = variables['api_post_valid_author']
+    # send the patch author requests with the developer role
+    delete_author = requests.delete(
+        url=f'{base_url}{_addon_create}{addon}/authors/{author}/',
+        headers={'Authorization': f'Session {session_auth}'},
+    )
+    assert (
+        delete_author.status_code == 403
+    ), f'Actual response: {delete_author.status_code}, {delete_author.text}'
+    assert (
+        'You do not have permission to perform this action.' in delete_author.text
+    ), f'Actual message was {delete_author.text}'
+
+
+@pytest.mark.serial
+@pytest.mark.create_session('staff_user')
+def test_addon_developer_role_cannot_delete_addon(base_url, session_auth):
+    """Verify that an addon cannot be deleted by an author with a developer role;
+    only owners are allowed to delete addons"""
+    addon = payloads.edit_addon_details['slug']
+    delete_addon = requests.get(
+        url=f'{base_url}{_addon_create}{addon}/delete_confirm/',
+        headers={'Authorization': f'Session {session_auth}'},
+    )
+    assert (
+        delete_addon.status_code == 403
+    ), f'Actual response: {delete_addon.status_code}, {delete_addon.text}'
+    assert (
+        'You do not have permission to perform this action.' in delete_addon.text
+    ), f'Actual message was {delete_addon.text}'
+
+
+@pytest.mark.serial
+@pytest.mark.create_session('staff_user')
+@pytest.mark.clear_session
+def test_addon_developer_role_can_request_author_details(
+    selenium, base_url, variables, session_auth
```

Comment:
do you need the ```selenium``` object here?

Reply:
The selenium fixture is currently handling the `clear_session` part, this is why I'm using it here. I agree it is not very intuitive to use selenium in a purely API test so I might try to decouple the `clear_session` part and make it its own fixture. I will do that in another PR though, because it will require changes in more places.